sycl/doc/SYCL2020-SpecializationConstants.md

```diff
@@ -0,0 +1,769 @@
+# Specialization constants
+
+Specialization constants are implemented in accordance with how they are defined
+by SYCL 2020 specification: [SYCL registry][sycl-registry],
+[direct link to the specification][sycl-2020-spec].
+
+[sycl-registry]: https://www.khronos.org/registry/SYCL/
+[sycl-2020-spec]: https://www.khronos.org/registry/SYCL/specs/sycl-2020/pdf/sycl-2020.pdf
```

Comment:
nit: I suggest replacing with a link to HTML version of the specification. It enables references to exact section in the document. For instance - https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_specialization_constants.

Reply:
Done in a0832d5d83ba1411522a02dc5c8cc5b95aca1fa1, thanks for the suggestion!